File: app/src/main/java/com/glucocontrol/domain/usecase/auth/SignInWithEmailUseCase.kt
package com.glucocontrol.domain.usecase.auth

import com.glucocontrol.domain.repository.AuthRepository
import javax.inject.Inject

class SignInWithEmailUseCase @Inject constructor(private val authRepository: AuthRepository) {
    suspend operator fun invoke(email: String, password: String) = authRepository.signInWithEmail(email, password)
}

File: lib/domain/usecase/auth/sign_in_with_email_use_case.dart
import 'package:glucocontrol/domain/repository/auth_repository.dart';

class SignInWithEmailUseCase {
  final AuthRepository _repo;
  SignInWithEmailUseCase(this._repo);
  Future<void> call(String email, String password) =>
      _repo.signInWithEmail(email, password);
}
